[WiP] New Job Name Validation

samatdav opened the pull request:
Improvement of the javascript client-side name validation.

olivergondza committed:
[FIXED JENKINS-33780] Introduce NodeListener

olivergondza committed:
Avoid log clutter

olivergondza committed:
[JENKINS-33780] Catch all throwables from implementations

varmenise committed:
[FIXED JENKINS-34883] the legacy default update site can be different from the one specified by ID_DEFAULT

varmenise committed:
[FIXED JENKINS-34883] use an ovverridable createDefaultUpdateSite method

varmenise committed:
[FIXED JENKINS-34883] make method protected

samatdav committed:
Improved new job name validation

samatdav committed:
delete space

kzantow committed:
JENKINS-34708 - UpdateCenter.EnableJob not properly setting status (#2365)

* JENKINS-34708 - UpdateCenter.EnableJob not properly setting status

* Restore if (status instanceof Pending)

samatdav committed:
removed focus and unneccesary client side check

samatdav committed:
new line at the end

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.6

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

orrc committed:
Fix unclosed <code> tag in the 2.6 changelog. (#2366)

varmenise committed:
Merge pull request #2363 from varmenise/JENKINS-34883

[FIXED JENKINS-34883] the legacy default update site can be different from the one specified by ID_DEFAULT

recena committed:
[JENKINS-35020] Fixed some JSHint errors (#2368)

* [JENKINS-35020] Fixed some JSHint errors

* [JENKINS-35020] @kzantow's comment was addressed

* [JENKINS-35020] @kzantow's comment was addressed

kzantow committed:
[FIX JENKINS-33947] - Fix keyboard navigation in setup wizard (#2294)

daspilker committed:
[JENKINS-34751] use patched version of Groovy 2.4.6 (#2369)

contains fixes for GROOVY-7761 and GROOVY-7826

[FIXES JENKINS-34751]

recena committed:
[JENKINS-34438] - Activate JSHint through js-builder (#2367)

madhead committed:
Update russian translations of nodes help pages to match latest English variant (#2376)

damianszczepanik committed:
Fixes Jenkins build error

damianszczepanik committed:
Updated Polish translations

damianszczepanik committed:
Merge pull request #2380 from damianszczepanik/build-error

Fixes Jenkins build error

damianszczepanik committed:
Merge pull request #2382 from damianszczepanik/translation-pl

Updated Polish translations

kamilszymanski committed:
Add symbol annotation for SystemInfoLink (#2375)

jglick committed:
[JENKINS-35190] Do not even consult PingFailureAnalyzer if ping fails on the agent side. (#2377)

peppelan committed:
[JENKINS-35198] - DelegatingComputerLauncher should accept child classes in its hooks (#2384)

* JENKINS-35198 Add failing test for clear identification of the issue and future regression test

* JENKINS-35198 Fix DelegatingComputerLauncher logic for filtering out subclasses as possible delegates

this logic is by default, and subclasses can still allow the selection of other DelegatingComputerLauncher instances as delegates.

damianszczepanik committed:
[FIXED JENKINS-35201] - NodeJS could not be built on Win x64 (#2381)

oleg-nenashev committed:
Noting #2368, #2294, #2381, #2384, #2377, #2375, #2380, #2367, #2369

samatdav committed:
sidepanel popup change

samatdav committed:
comment fix

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.7

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

olivergondza committed:
State label name as well in order to disambiguate when used from Form… (#2349)

* State label name as well in order to disambiguate when used from FormValidation#aggregate

recena committed:
[JENKINS-35178] Fix close button in Safari (#2378)

* [JENKINS-35178] Fix close button in Safari

* [JENKINS-35178] Removed units

* [JENKINS-35178] Back to the original button design

samatdav committed:
dirty popup with library

PeterDaveHello committed:
optimize png images losslessly using zopflipng (#2379)

Optimize PNG images losslessly using zopflipng

tomass committed:
Lithuanian translation update (#2359)

* Lithuanian translation update

* Lithuanian translation update

kzantow committed:
[FIXED JENKINS-34881] - Handle pre-configured security settings for new installs (#2364)

* [FIXED JENKINS-34881] - handle non-default security settings for new installs

* Ensure permissions

* Initial security authentication token should still follow redirects

kohsuke committed:
Update windows slave installer module from 1.5.1 to 1.6 (#2386)

See https://github.com/jenkinsci/windows-slave-installer-module/pull/3

oleg-nenashev committed:
[JENKINS-35206] - Make the InstallState object deserialization robust against corrupted files (#2387)

* [JENKINS-35206] - Add unit tests for the deserialization logic

* [JENKINS-35206] - Install state should be robust against messed statuses when deserializing objects

* [JENKINS-35206] - Fix the license header in thre test file

escoem committed:
[FIXED JENKINS-31915] Proxy settings in plugins page are ignored (#1955)

[FIXED JENKINS-31915] Proxy settings in plugins page are ignored

kohsuke committed:
Removed code duplication

getItem() and contains() act exactly the same

olivergondza committed:
Merge pull request #2314 from olivergondza/node-listener

[FIXED JENKINS-33780] Introduce listener for slave creation/update/deletion

olivergondza committed:
Noting #2314

samatdav committed:
Removed unnecessary st:include tags

samatdav committed:
remove st:include

kzantow committed:
[JENKINS-35206] - fix potential NPE due to corrupt Jenkins.InstallState (#2385)

oleg-nenashev committed:
Noting #2378, #2379, #2364, #2387, #1955 and #2385

oleg-nenashev committed:
Copy-editing of the current changelog entries

oleg-nenashev committed:
Noting #2367, which was missing in 2.6

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.8

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

oleg-nenashev committed:
Fix the link formatting in the 2.8 changelog

kohsuke committed:
Merge pull request #2389 from jenkinsci/queue-contains

Removed code duplication

kohsuke committed:
This appears in javadoc

Commit 7efb9090ddb6d63911e388cdf90e5f42dd880da3 seems to have done
without paying attention to what javadoc refers to.

kohsuke committed:
Typo

stephenc committed:
[FIXED JENKINS-35402] Make it easier for update sites to tweak the InstallationJob

ndeloof committed:
[JENKINS-34923]

Signed-off-by: Nicolas De Loof <nicolas.deloof@gmail.com>

daniel-beck committed:
[FIX JENKINS-35381] Restrict ResourceBundleUtil (#2393)

stephenc committed:
Merge pull request #2394 from stephenc/custome-installation-job

[FIXED JENKINS-35402] Make it easier for update sites to tweak the InstallationJob

daspilker committed:
updated Groovy to 2.4.7

follow-up for JENKINS-34751

samatdav committed:
Added newJob properties to sidepanel properties to fix labels

samatdav committed:
uncomment popup

samatdav committed:
romoved Bootstrap from newJob

samatdav committed:
romoved unnecessary popup jelly

samatdav committed:
romoved unnecessary popup jelly

samatdav committed:
changed popup.css

olivergondza committed:
Merge pull request #2399 from daspilker/JENKINS-34751

[JENKINS-34751] updated Groovy to 2.4.7

samatdav committed:
put remodal popup styles to remodal.css

samatdav committed:
added remodal.css

ndeloof committed:
some obvious javadoc because some of us feel method names are not enough

ndeloof committed:
Merge pull request #2395 from ndeloof/one-shot-6

[JENKINS-34923]

samatdav committed:
moved my rule to addform.less

samatdav committed:
moved popup from sidepanel.jelly to popup.jelly

jglick committed:
[JENKINS-35098] Disable AutoBrowserHolder by default to improve the changelog rendering performance (#2371)

* [FIXED JENKINS-35098] Deleting AutoBrowserHolder.

* Normalizing form binding scheme for AbstractProject.scm.

* At @oleg-nenashev’s insistence, restoring AutoBrowserHolder, though not using it by default.

* Using SystemProperties at @oleg-nenashev’s recommendation.

Commit:
[FIXED JENKINS-35641] - Always send usage statistics over HTTPs to the new usage.jenkins.io hostname(#2398)

References INFRA-559

oleg-nenashev committed:
Update remoting to 2.60 (#2403)

Changes summary:

Fixed issues:
* [JENKINS-22722](https://issues.jenkins-ci.org/browse/JENKINS-22722) - 
Make the channel reader tolerant against Socket timeouts. 
(https://github.com/jenkinsci/remoting/pull/80)
* [JENKINS-32326](https://issues.jenkins-ci.org/browse/JENKINS-32326) - 
Support no_proxy environment variable. 
(https://github.com/jenkinsci/remoting/pull/84)
* [JENKINS-35190](https://issues.jenkins-ci.org/browse/JENKINS-35190)  - 
Do not invoke PingFailureAnalyzer for agent=>master ping failures. 
(https://github.com/jenkinsci/remoting/pull/85)
* [JENKINS-31256](https://issues.jenkins-ci.org/browse/JENKINS-31256) - 
 <code>hudson.Remoting.Engine#waitForServerToBack</code> now uses credentials for connection. 
(https://github.com/jenkinsci/remoting/pull/87)
* [JENKINS-35494](https://issues.jenkins-ci.org/browse/JENKINS-35494) - 
Fix issues in file management in <code>hudson.remoting.Launcher</code> (main executable class). 
(https://github.com/jenkinsci/remoting/pull/88)

Enhancements:
* Ensure a message is logged if remoting fails to override the default <code>ClassFilter</code>. 
(https://github.com/jenkinsci/remoting/pull/80)

oleg-nenashev committed:
Noting #2393, #2394, #2399, #2395, #2371, #2398, #2403

bwalding committed:
Make text gender neutral (#2405)

samatdav committed:
changed the look of new job creation

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.9

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

daniel-beck committed:
Fix links to JIRA

samatdav committed:
Revert "Improved new job name validation"

This reverts commit ed80c35d3bcc1dd1230a48116b293f4bc94ae3b9.

samatdav committed:
Revert "delete space"

This reverts commit d2dff11357886105edf141940a49607dacd6018a.

samatdav committed:
commiting changes

samatdav committed:
Revert "Revert "Improved new job name validation""

This reverts commit 4a3f75ac675499f8058956b2ad2aa1217d10e78d.

samatdav committed:
commit files

samatdav committed:
removed UI changes

oleg-nenashev committed:
[JENKINS-33600] - User#isIdOrFullnameAllowed() should be tolerant against null parameters (#2406)

* [JENKINS-33600] - User#isIdOrFullnameAllowed() should be tolerant against null parameters

* [JENKINS-33600] - Add the follow-up TODO

Commit:
Better exception message if a SecurityRealm doesn't respect the API contract and return null (#2407)

* Getter exception message if the SecurityRealm doesn't respect the API contract and return null.

* Use NullPointerException instead of IllegalStateException to not change the behaviour of the method.

samatdav committed:
moved properties and removed name validation change

samatdav committed:
fixed mistakes

samatdav committed:
removed task creation on new page

samatdav committed:
check on keyup

samatdav commented:
@lanwen Could you help understand what happens at `$('#items').data('valid')`? [In add-item.js](https://github.com/samatdav/jenkins/pull/3/files#diff-e146d4fdbfe6385a456aea9775f6282dR53) line 53 it is called to enable submit button if it returns `true`. Strangely, `$('#items').data('valid')` always returns `undefined` in my popup. However, it works in /newJob page. Moreover, validation of name `$('#name').data('valid')` works well in my popup as in /newJob.
I am confused since it used to work well recently. I could not find any changes of mine that could cause it. I did `git checkout old-commit` and it stopped working there also. However, it worked fine at that commit

superboum committed:
Add compilation support for 32 bits linux - node.js fix (#2414)

oleg-nenashev committed:
Noting #2414, #2407 and #2406

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.10

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

samatdav committed:
removed popup

samatdav committed:
removed popup fix

samatdav committed:
different handeling of keyup and blur

lanwen commented:
You can point to exact line like here (https://github.com/samatdav/jenkins/blob/74ba052339388baf635831bca3ff8ddeba27c038/war/src/main/js/add-item.js#L53)

`$('#items').data('valid')` means that element with id `items` should have attr `data-valid` with some value.
So just check what elements have such id and what sets this `data-valid` attr.

stephenc committed:
[FIXED JENKINS-36123] Add an extension point that allows plugins to veto polling

stephenc committed:
[JENKINS-36123] Address review comments from James Nord

stephenc committed:
[JENKINS-36123] Refactor new extension point name

aldaris committed:
[FIXED JENKINS-35981] Allow Setup wizard to work on Tomcat 8 (#2411)

stephenc committed:
Merge pull request #2418 from stephenc/jenkins-36123

[FIXED JENKINS-36123] Add an extension point that allows plugins to veto polling

stephenc committed:
Noting merge of JENKINS-36123

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.11

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

stephenc committed:
[FIXED JENKINS-36280] Enable DescriptorVisibilityFilter for Slave's ComputerLauncher, RetentionStrategy and NodeProperty

kohsuke committed:
Make BulkChange auto-closeable

so that it can be used as below:

    try (BulkChange bc = new BulkChange(o)) {
       ...
       bc.commit();
    }

stephenc committed:
[JENKINS-36280] Address review comments

vjuranek committed:
[FIXED JENKINS-36277] Check that process working dir exists (#2424)

Commit:
[JENKINS-32027]  Avoiding to refresh codemirror through the layoutUpdateCallback (#2390)

stephenc committed:
Tests did not match naming pattern so were never being executed

tn-osimis committed:
Update BUILD_TAG description to mention the replacement of slashes with dashes (#2417)

Divergence introduced with 117d69b8164872ba22f5ceb46b60f2cae23e29db.

stephenc committed:
[JENKINS-36280] Add some tests

The tests pass locally, but they will not run until #2429 is merged

stephenc committed:
Merge pull request #2429 from stephenc/enable-tests

Tests did not match naming pattern so were never being executed

stephenc committed:
Merge branch 'master' into jenkins-36280

jglick committed:
Stray import.

mslusarczyk committed:
[JENKINS-21486] Fixing optional dependency version resolution. (#2421)

Without it version of optional dependencies is set to "X.Y;resolution:=optional" which causes problems when using hudson.util.VersionNumber comparison methods.

kohsuke committed:
Merge pull request #2428 from jenkinsci/closeable-bulkchange

Make BulkChange auto-closeable

ThomasGP committed:
[FIXED JENKINS-36232] NPE during SCM polling (#2425)

* [FIXED JENKINS-36232] NPE during SCM polling

* add test to trigger the "no veto" code path that has the NPE

* incorporate remark on using @Issue reference

jglick committed:
[JENKINS-35906] Ensure that SCMDescriptor.newInstance overrides are honored (#2426)

* [FIXED JENKINS-35906] Ensure that SCMDescriptor.newInstance overrides are honored.

* [FIXED JENKINS-36043] Work around fragile form submission design in multi-branch-project-plugin.

stephenc committed:
Merge pull request #2427 from stephenc/jenkins-36280

[FIXED JENKINS-36280] Enable DescriptorVisibilityFilter for Slave's C…

stephenc committed:
Noting merge of JENKINS-36280

alvarolobato committed:
[JENKINS-20187] - Handle growing files when creating a tar file. (#2419)

* [JENKINS-20187] - Handle growing files when creating a tar file.

* - Introduce try-with-resources
- Remove exception login
- Throw new exception with filename and addSuppresed

* Address jglick comments

olivergondza committed:
[FIXED JENKINS-25416][JENKINS-28790] Do not inject build variables into maven process optionally (#1976)

* [FIXED JENKINS-25416][JENKINS-28790] Do not inject build variables into maven process optionally

* [JENKINS-25416][JENKINS-28790] Fixed typo in test method name

* [JENKINS-25416][JENKINS-28790] Suggest alternatives in help

* [JENKINS-25416][JENKINS-28790] Use less hackish approach to preserving legacy behaviour

oleg-nenashev committed:
Noting #2417, #2428, #2424, #2390, #2421, #2425, #2419, #1976

oleg-nenashev committed:
[JENKINS-25416][JENKINS-28790] - Fix the parametersReferencedFromPropertiesShouldRetainBackslashes() test, which relies on the legacy behavior (#2435)

The test fails after https://github.com/jenkinsci/jenkins/pull/1976/commits/84ee34f2df620f1280022f1678b81e5c27dfc330

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-2.12

kohsuke committed:
updated changelog for release

oleg-nenashev committed:
Annotate DescriptorList methods and add Javadocs (#2434)

samatdav committed:
Merge remote-tracking branch 'upstream/master' into new-job-name-validation

samatdav committed:
fix JSHint errors

samatdav committed:
cleanup

samatdav committed:
added single quote

samatdav committed:
separated button enabling and focus

samatdav committed:
cleanup

samatdav committed:
Tom's suggestions

samatdav committed:
moved invalid warning

samatdav committed:
red border

samatdav committed:
red background

lanwen commented on war/src/main/js/add-item.js:
```diff
@@ -247,10 +247,10 @@ $.when(getItems()).done(function(data) {
           if (message !== '') {
             activateValidationMessage('#itemname-invalid', '.add-item-name', message);
-            $('#add-item-panel input').css('border', "1px solid red");
+            $('input[name="name"]').css({'border': "1px solid red", "background-color": "#FADEDE"});
```
should toggle class, not css

samatdav committed:
class toggle

samatdav committed:
renamed class
